dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,214 @@
+    @JSONP
+    @NoCache
+    @Produces(SseFeature.SERVER_SENT_EVENTS)
+    public final EventOutput getLogs(@Context final HttpServletRequest request,
+            @PathParam("fileName") final String fileName) throws IOException {
+
+        new WebResource().init(null, request, new EmptyHttpResponse(), true, null);
+
+        if(fileName.trim().isEmpty()) {
+            return sendError("File should not be empty");
+        }
+
+
+        String tailLogLofFolder = com.dotmarketing.util.Config.getStringProperty("TAIL_LOG_LOG_FOLDER", "./dotsecure/logs/");
+        if (!tailLogLofFolder.endsWith(java.io.File.separator)) {
+            tailLogLofFolder = tailLogLofFolder + java.io.File.separator;
+        }
+
+        final File logFolder 	= new File(com.dotmarketing.util.FileUtil.getAbsolutlePath(tailLogLofFolder));
+        final File logFile 	= new File(com.dotmarketing.util.FileUtil.getAbsolutlePath(tailLogLofFolder + fileName));
+
+
+        // if the logFile is outside of the logFolder, die
+        if ( !logFolder.exists()
+                ||   !logFile.getCanonicalPath().startsWith(logFolder.getCanonicalPath())) {
+
+            final String errorMessage = "Invalid File request:" + logFile.getCanonicalPath() + " from:" +request.getRemoteHost();
+            SecurityLogger.logInfo(
+                    TailLogResource.class,  errorMessage);
+            return sendError(errorMessage);
+        }
+
+        Logger.info(this.getClass(), "Requested logFile:" + logFile.getCanonicalPath());
+
+        final String regex = Config.getStringProperty("TAIL_LOG_FILE_REGEX", ".*\\.log$|.*\\.out$");
+
+
+        if(!Pattern.compile(regex).matcher(fileName).matches()){
+            sendError("Error ");
+        }
+
+        final MyTailerThread thread = new MyTailerThread(logFile);
+
+        String name = null;
+        for (int i = 0; i < 1000; i++) {
+            name = "LogTailer" + i + ":" + fileName;
+            Thread t = ThreadUtils.getThread(name);
+            if (t == null) {
+                break;
+            }
+            if (i > 100) {
+                sendError("Too many Logger threads");
+            }
+        }
+
+        thread.setName(name);
+
+        thread.start();
+
+        return thread.getEventOutput();
+    }
+
+    private EventOutput sendError(final String errorMessage){
+
+        final EventOutput eventOutput = new EventOutput();
+
+        final OutboundEvent.Builder eventBuilder = new OutboundEvent.Builder();
+        eventBuilder.name("failure");
+        eventBuilder.data(Map.class,
+                map("failure", errorMessage));
+        final OutboundEvent event = eventBuilder.build();
+        try {
+            eventOutput.write(event);
+        } catch (Exception e1) {
+            throw new DotRuntimeException(e1);
+        }
+
+        return eventOutput;
+    }
+
+
+    private class MyTailerListener extends TailerListenerAdapter {
+
+        StringWriter out = new StringWriter();
```

Comment:
May be a StringBuffer would be easier since is already sync

Reply:
Done